inference-engine/tests/functional/plugin/shared/src/behavior/dynamic_batch.cpp

```diff
@@ -0,0 +1,165 @@
+                const auto &info = input.second;
+                auto blob = GenerateInput(*info);
+                inputs.push_back(blob);
+            }
+            reference_inputs.push_back(inputs);
+            reference_outputs.push_back(CalculateRefs());
+        }
+
+        for (int i = 0; i < batch_sizes.size(); i++) {
+            infer_requests.push_back(executableNetwork.CreateInferRequest());
+            auto batch_size = batch_sizes[i];
+
+            auto& infer_request = infer_requests[i];
+            infer_request.SetBatch(batch_size);
+
+            inputs.clear();
+            for (const auto &input : executableNetwork.GetInputsInfo()) {
+                const auto &info = input.second;
+                auto blob = GenerateInput(*info);
+                infer_request.SetBlob(info->name(), blob);
+                inputs.push_back(blob);
+            }
+
+            scaled_inputs.push_back(inputs);
+
+            for (int j = 0; j < reference_inputs[i].size(); j++) {
+                auto& ref = reference_inputs[i][j];
+                auto& actual = scaled_inputs[i][j];
+
+                auto byte_num = ref->byteSize();
+                auto ref_ptr = ref->buffer().as<uint8_t*>();
+                auto actual_ptr = actual->buffer().as<uint8_t*>();
+
+                for (int k = 0; k < byte_num; k++) {
+                    actual_ptr[k] = ref_ptr[k];
+                }
+            }
+        }
+
+        for (auto& infer_request : infer_requests) {
+            if (run_async) {
+                infer_request.StartAsync();
+            } else {
+                infer_request.Infer();
+            }
+        }
+
+        if (run_async) {
+            for (auto& infer_request : infer_requests) {
+                auto status = infer_request.Wait(10000);
+                if (status != InferenceEngine::StatusCode::OK) {
+                    GTEST_FAIL() << "Inference request status after wait is not OK";
+                }
+            }
+        }
+    }
+
+    void DynamicBatchTest::Validate() {
+        for (int i = 0; i < infer_requests.size(); i++) {
+            auto outputs = std::vector<InferenceEngine::Blob::Ptr>{};
+            for (const auto &output : executableNetwork.GetOutputsInfo()) {
+                const auto &name = output.first;
+                outputs.push_back(infer_requests[i].GetBlob(name));
+            }
+            for (int j = 0; j < reference_outputs[i].size(); j++) {
+                if (reference_outputs[i][j].size() < outputs[j]->byteSize()) {
+                    auto actual_ptr = outputs[j]->buffer().as<uint8_t*>();
+                    for (int k = reference_outputs[i][j].size(); k < outputs[j]->byteSize(); k++) actual_ptr[k] = 0;
+                    reference_outputs[i][j].resize(outputs[j]->byteSize());
+                }
+            }
+            Compare(reference_outputs[i], outputs);
+        }
+    }
+
+    void DynamicBatchTest::Run() {
```

Comment:
This one is needed since LoadNetwork is not virtual and cannot be overloaded, hence original Run will execute wrong function